internal/mode/static/nginx/agent/command.go

```diff
@@ -68,73 +126,347 @@ func (cs *commandService) Subscribe(in pb.CommandService_SubscribeServer) error
 		return agentgrpc.ErrStatusInvalidConnection
 	}
 
-	cs.logger.Info(fmt.Sprintf("Received subscribe request from %q", gi.IPAddress))
-
-	go cs.listenForDataPlaneResponse(ctx, in)
-
-	// wait for the agent to report itself
-	podName, err := cs.waitForConnection(ctx, gi)
+	// wait for the agent to report itself and nginx
+	conn, deployment, err := cs.waitForConnection(ctx, gi)
 	if err != nil {
 		cs.logger.Error(err, "error waiting for connection")
 		return err
 	}
 
-	cs.logger.Info(fmt.Sprintf("Handling subscription for %s/%s", podName, gi.IPAddress))
+	cs.logger.Info(fmt.Sprintf("Successfully connected to nginx agent %s", conn.PodName))
+
+	msgr := messenger.New(in)
+	go msgr.Run(ctx)
+
+	// apply current config before starting event loop
+	deployment.Lock.RLock()
+	if err := cs.setInitialConfig(ctx, deployment, conn, msgr); err != nil {
+		deployment.Lock.RUnlock()
+
+		return err
+	}
+
+	// subscribe to the deployment broadcaster to get file updates
+	broadcaster := deployment.GetBroadcaster()
+	channels := broadcaster.Subscribe()
+	defer broadcaster.CancelSubscription(channels.ID)
+	deployment.Lock.RUnlock()
```

Comment:
Can you unlock before creating the broadcaster and channels? I feel like the farther the Unlock is from the Lock the riskier it is.